packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,388 @@
+    
+         - data: splitted string array (frame values), values are shift()ed so
+         this should be empty after parsing the whole hierarchy.
+         - frameTime: playback time for this keyframe.
+         - bone: the bone to read frame data from.
+
+         Note: Position data (specifically Z) is flipped to convert coordinate systems.
+    */
+    private static _ReadFrameData(data: string[], frameTime: number, bone: BVHNode) {
+        if (bone.type === "ENDSITE")
+            // end sites have no motion data
+            return;
+
+        // add keyframe
+        const keyframe = new BVHKeyFrame();
+        keyframe.time = frameTime;
+        keyframe.position = new Vector3();
+        keyframe.rotation = new Quaternion();
+
+        bone.frames.push(keyframe);
+
+        let pitch = 0,
+            yaw = 0,
+            roll = 0;
+
+        // parse values for each channel in node
+        for (let i = 0; i < bone.channels.length; ++i) {
+            const value = data.shift();
+            if (!value) {
+                continue;
+            }
+
+            switch (bone.channels[i]) {
+                case this._XPosition:
+                    keyframe.position.x = parseFloat(value.trim());
+                    break;
+                case this._YPosition:
+                    keyframe.position.y = parseFloat(value.trim());
+                    break;
+                case this._ZPosition:
+                    keyframe.position.z = -parseFloat(value.trim()); // Flip Z position
+                    break;
+                case this._XRotation:
+                    pitch = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                case this._YRotation:
+                    yaw = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                case this._ZRotation:
+                    roll = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                default:
+                    throw new Error("invalid channel type");
+            }
+        }
+
+        if (yaw != 0 || pitch != 0 || roll != 0) {
+            // Create rotation matrix in proper order
+            const rotationMatrix = Matrix.Identity();
+            Matrix.RotationYawPitchRollToRef(yaw, pitch, roll, rotationMatrix);
+            keyframe.rotation.fromRotationMatrix(rotationMatrix);
+        }
+
+        // parse child nodes
+        for (let i = 0; i < bone.children.length; ++i) {
+            BVHParser._ReadFrameData(data, frameTime, bone.children[i]);
+        }
+    }
+}
+
+class LoaderContext {
+    animationName: string = "";
+    loopMode: number = Animation.ANIMATIONLOOPMODE_CYCLE;
+    list: BVHNode[] = [];
+    root: BVHNode = new BVHNode();
+    numFrames: number = 0;
+    frameRate: number = 0;
+    skeleton: Skeleton;
+
+    constructor(skeleton: Skeleton) {
+        this.skeleton = skeleton;
+    }
+}
+
+class BVHNode {
```

Comment:
Are these classes needed? Can't they be a const that satisfies an interface? would be great to avoid the class creation overhead where it is not needed.

Reply:
referring to classes without functions or constructor.